apps/vc-api/src/middlewares/http-logger.middleware.ts

```diff
@@ -12,6 +12,27 @@ export class HttpLoggerMiddleware implements NestMiddleware {
 
     let finished = false;
 
+    const oldWrite = res.write;
+    const oldEnd = res.end;
+    const chunks: Buffer[] = [];
+    let responseBody: string;
+
+    res.write = (...args: unknown[]): boolean => {
+      chunks.push(args[0] as Buffer);
+      return oldWrite.apply(res, args);
+    };
+
+    res.end = (...args: unknown[]) => {
+      const chunk = args[0] as Buffer;
+
+      if (chunk) {
+        chunks.push(chunk);
+      }
+
+      responseBody = Buffer.concat(chunks).toString();
+      return oldEnd.apply(res, args);
+    };
```

Comment:
This code seems really safe, but maybe it's better to be extra careful and have a try/catch as we don't want the logging code to break the application, IMO.
```suggestion
    res.write = (...args: unknown[]): boolean => {
      try {
        chunks.push(args[0] as Buffer);
      } catch {
        this.logger.warn("unable to log response")
      }
      return oldWrite.apply(res, args);
    };

    res.end = (...args: unknown[]) => {
      try {
        const chunk = args[0] as Buffer;
        if (chunk) {
          chunks.push(chunk);
        }
        responseBody = Buffer.concat(chunks).toString();
      } catch {
        this.logger.warn("unable to log response")
      }
      return oldEnd.apply(res, args);
    };
```

Reply:
good point!